src/devices/screen.c

```diff
@@ -0,0 +1,328 @@
+/**
+ * \file screen.c
+ *
+ * \brief Brain screen display and touch functions.
+ *
+ * Contains user calls to the v5 screen for touching and displaying graphics.
+ *
+ * \copyright (c) 2017-2021, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "pros/screen.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/_stdint.h>
+
+#include "common/linkedlist.h"
+#include "kapi.h"
+#include "pros/apix.h"
+#include "v5_api.h"  // vexDisplay* 
+
+/******************************************************************************/
+/**                  Screen Graphical Display Functions                      **/
+/**                                                                          **/
+/**   These functions allow programmers to display shapes on the v5 screen   **/
+/******************************************************************************/
+
+static mutex_t _screen_mutex = NULL;
+
+typedef struct touch_event_position_data_s {
+	int16_t x;
+	int16_t y;
+} touch_event_position_data_s_t;
+
+void screen_set_pen(uint32_t color){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayForegroundColor(color);
+	mutex_give(_screen_mutex);
+}
+
+void screen_set_eraser(uint32_t color){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayBackgroundColor(color);
+	mutex_give(_screen_mutex);
+}
+
+uint32_t screen_get_pen(void){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	uint32_t color = vexDisplayForegroundColorGet();
+	mutex_give(_screen_mutex);
+	return color;
+}
+
+uint32_t screen_get_eraser(void){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	uint32_t color = vexDisplayBackgroundColorGet();
+	mutex_give(_screen_mutex);
+	return color;
+}
+
+void screen_clear(void){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayErase();
+	mutex_give(_screen_mutex);
+}
+
+void screen_scroll(int16_t start_line, int16_t lines){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayScroll(start_line, lines);
+	mutex_give(_screen_mutex);
+}
+
+void screen_scroll_area(int16_t x0, int16_t y0, int16_t x1, int16_t y1, int16_t lines){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayScrollRect(x0, y0, x1, y1, lines);
+	mutex_give(_screen_mutex);
+}
+
+void screen_copy_area(int16_t x0, int16_t y0, int16_t x1, int16_t y1, uint32_t* buf, int32_t stride){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayCopyRect(x0, y0, x1, y1, buf, stride);
+	mutex_give(_screen_mutex);
+}
+
+void screen_draw_pixel(int16_t x, int16_t y){
+	mutex_take(_screen_mutex, TIMEOUT_MAX);
+	vexDisplayPixelSet(x, y);
+	mutex_give(_screen_mutex);
+}
+
+void screen_clear_pixel(int16_t x, int16_t y){
```

Comment:
Change all instances of "clear" in api to "erase" because we have "set_eraser" and in the background that's pretty much all that it's doing (clearing = setting the color to the eraser color)